### PR TITLE
Packages: Move media to new structure 

### LINKED ISSUE #7826: Packages: Move `media` to new structure
## Task Description

**Context:**

Media related utils need to be moved into their own package so they can be used by story-editor, dashboard, and wp-editor without any circular dependencies.

See https://github.com/google/web-stories-wp/tree/try/next-level-packages/packages/media for inspiration / proof-of-concept.

**To-Do:**

* [ ] Move any media-related utils and constants (See PoC above) + tests to `packages/media/src`
* [ ] Update imports where necessary.  
  Use `@web-stories-wp/...` imports for anything that is already a package, relative imports for everything else.
* [ ] Create `packages/media/package.json` referencing its dependencies
* [ ] Create `packages/media/README.md` with short documentation.
* [ ] Bonus: add tests for utils that don't have any yet
* [ ] Bonus: add JSDoc type definitions, and add JSDoc do utils which don't have any yet

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Media related utils need to be moved into their own package so they can be used by story-editor, dashboard, and wp-editor without any circular dependencies.

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7826

### CODE REVIEW

File: assets/src/edit-story/elements/media/index.js [hunk 1]
```diff
@@ -24,7 +24,6 @@ import styled from 'styled-components';
  */
 import { elementWithBorderRadius } from '../shared';
 
-export { default as getMediaSizePositionProps } from './getMediaSizePositionProps';
 export { default as getFocalFromOffset } from './getFocalFromOffset';
```
Comment (miina) [inline assets/src/edit-story/elements/media/index.js]: Should this also be moved?

File: assets/src/edit-story/app/media/utils/useMediaUploadQueue/index.js [hunk 2]
```diff
@@ -23,20 +23,22 @@ import {
...
  */
 import { useUploader } from '../../../uploader';
 import useReduction from '../../../../utils/useReduction';
-import { createBlob } from '../../../../utils/blobs';
 import { noop } from '../../../../utils/noop';
-import getFileName from '../getFileName';
 import useUploadVideoFrame from '../useUploadVideoFrame';
 import useFFmpeg from '../useFFmpeg';
 import getResourceFromAttachment from '../getResourceFromAttachment';
 import getResourceFromLocalFile from '../getResourceFromLocalFile';
```
Comment (miina) [inline assets/src/edit-story/app/media/utils/useMediaUploadQueue/index.js:41]: ^ Should these also be moved?

File: assets/src/edit-story/app/media/utils/index.js [hunk 3]
```diff
@@ -14,13 +14,9 @@
  * limitations under the License.
  */
 
-export { default as createResource } from './createResource';
-export { default as getFirstFrameOfVideo } from './getFirstFrameOfVideo';
 export { default as getResourceFromAttachment } from './getResourceFromAttachment';
 export { default as getResourceFromLocalFile } from './getResourceFromLocalFile';
 export { default as getResourceFromMediaPicker } from './getResourceFromMediaPicker';
 export { default as getResourceFromMedia3p } from './getResourceFromMedia3p';
```
Comment (miina) [inline assets/src/edit-story/app/media/utils/index.js:20]: ^ Should these also be moved?

File: packages/media/src/test/getResourceSize.js [hunk 4]
```diff
@@ -18,7 +18,10 @@
  * Internal dependencies
  */
 import getResourceSize from '../getResourceSize';
-import { FULLBLEED_HEIGHT, PAGE_WIDTH } from '../../../../constants';
+import {
+  FULLBLEED_HEIGHT,
+  PAGE_WIDTH,
+} from '../../../../assets/src/edit-story/constants';
```
Comment (miina) [inline packages/media/src/test/getResourceSize.js]: I guess it would be good to get the units package merged first.

**Review by miina — COMMENTED** [review]
Looks like it would be good to merge the units package first and then remove the references to `edit-story` inside the media package, too.

File: packages/media/src/test/getResourceSize.js (hunk 4 above)
Comment (spacedmonkey) [reply packages/media/src/test/getResourceSize.js]: I am hoping to merge unit package before this and rebase.

File: assets/src/edit-story/app/media/utils/index.js (hunk 3 above)
Comment (spacedmonkey) [reply assets/src/edit-story/app/media/utils/index.js:20]: Good question. In @swissspidy POC didn't bother. We can always move them later, but these are not used outside of editor.

File: assets/src/edit-story/app/media/utils/useMediaUploadQueue/index.js (hunk 2 above)
Comment (spacedmonkey) [reply assets/src/edit-story/app/media/utils/useMediaUploadQueue/index.js:41]: See https://github.com/google/web-stories-wp/pull/7954#discussion_r651631899

File: assets/src/edit-story/elements/media/index.js (hunk 1 above)
Comment (spacedmonkey) [reply assets/src/edit-story/elements/media/index.js]: Fixed.

File: packages/media/src/getResourceSize.js [hunk 5]
```diff
@@ -17,7 +17,10 @@
 /**
  * Internal dependencies
  */
-import { FULLBLEED_HEIGHT, PAGE_WIDTH } from '../../../constants';
+import {
+  FULLBLEED_HEIGHT,
+  PAGE_WIDTH,
+} from '../../../assets/src/edit-story/constants';
```
Comment (miina) [inline packages/media/src/getResourceSize.js]: This should now come from the units package, no?

Comment (miina) [reply packages/media/src/getResourceSize.js]: ^ My bad, was looking at an old diff, all good.

**Review by miina — APPROVED** [review]

**Pull request merged**